datumaro/plugins/icdar_format/extractor.py

```diff
@@ -0,0 +1,261 @@
+                        groups.append(group)
+                        colors.append((int(objects[0]), int(objects[1]), int(objects[2])))
+                        char = objects[9]
+                        if char[0] == '\"' and char[-1] == '\"':
+                            char = char[1:-1]
+                        chars.append(char)
+
+            mask_categories = MaskCategories({i: colors[i] for i in range(len(colors))})
+            mask_categories.inverse_colormap # pylint: disable=pointless-statement
+            gt_path = osp.join(_path, item_id + '_GT' + IcdarPath.GT_EXT)
+            if osp.isfile(gt_path):
+                inverse_cls_colormap = mask_categories.inverse_colormap
+                mask = lazy_mask(gt_path, inverse_cls_colormap)
+                # loading mask through cache
+                mask = mask()
+                classes = np.unique(mask)
+                for label_id in classes:
+                    if label_id != 0:
+                        image = self._lazy_extract_mask(mask, label_id)
+                        i = int(label_id)
+                        annotations.append(Mask(image=image, label=i, group=groups[i],
+                            attributes={ 'color': colors[i], 'char': chars[i],
+                            'center': centers[i] }))
+
+        return items
+
+    @staticmethod
+    def _lazy_extract_mask(mask, c):
+        return lambda: mask == c
+
+class _IcdarExtractor(SourceExtractor):
+    _TASK_EXTRACTORS = {
+        IcdarTask.word_recognition: _WordRecognitionExtractor,
+        IcdarTask.text_localization: _TextLocalizationExtractor,
+        IcdarTask.text_segmentation: _TextSegmentationExtractor,
+    }
+
+    def __init__(self, path, task):
+        if task is IcdarTask.word_recognition:
+            if not osp.isfile(path):
+                raise Exception("Can't read annotation file '%s'" % path)
+            subset = osp.basename(osp.dirname(path))
+            self._dataset_dir = osp.dirname(osp.dirname(path))
+        elif task is IcdarTask.text_localization or \
+                task is IcdarTask.text_segmentation:
+            if not osp.isdir(path):
+                raise Exception( "Can't open folder with annotation files'%s'" % path)
+            subset = osp.basename(path)
+            self._dataset_dir = osp.dirname(path)
+        self._path = path
+        self._task = task
+        super().__init__(subset=subset)
+
+        task_extractor = self._make_task_extractor(task)
+        self._categories = task_extractor.load_categories(self._dataset_dir,
+            self._path)
+        self._items = list(task_extractor.load_items(self._path, self._subset,
+            self._categories).values())
+
+    def _make_task_extractor(self, task):
+        if task not in self._TASK_EXTRACTORS:
+            raise NotImplementedError()
+        return self._TASK_EXTRACTORS[task]()
+
+class IcdarWordRecognitionExtractor(_IcdarExtractor):
```

Comment:
What is the point in classes above? We could put their methods in these classes and just call them in the base class.